### PR TITLE
Update Dockerfile syntax

### PR DESCRIPTION
**Pull request by rthallisey**
Make a few changes to the Dockerfiles to be more efficient when building.

### CODE REVIEW

File: apb-base/Dockerfile [hunk 1]
```diff
@@ -1,41 +1,38 @@
...
-COPY oc-login.sh /usr/bin
-COPY entrypoint.sh /usr/bin
+    && pip install -r requirements.txt \
+    && python setup.py develop \
+    && source ../ansible/hacking/env-setup
+RUN openshift-ansible-gen modules -o /usr/share/ansible/openshift
+
+RUN echo "localhost ansible_connection=local" > /etc/ansible/hosts \
+    && echo '[defaults]' > /etc/ansible/ansible.cfg \
+    && echo 'roles_path = /opt/ansible/roles' > /etc/ansible/ansible.cfg \
```
Comment (fabianvf) [inline apb-base/Dockerfile]: This needs to append (`>>`) or it will overwrite the [defaults] header

Comment (rthallisey) [reply apb-base/Dockerfile]: Looks like ```>``` got dropped when I rebased. Thanks!

File: apb-base/Dockerfile [hunk 2]
```diff
@@ -1,41 +1,38 @@
...
+RUN openshift-ansible-gen modules -o /usr/share/ansible/openshift
+
+RUN echo "localhost ansible_connection=local" > /etc/ansible/hosts \
+    && echo '[defaults]' > /etc/ansible/ansible.cfg \
+    && echo 'roles_path = /opt/ansible/roles' >> /etc/ansible/ansible.cfg \
+    && echo 'library = /usr/share/ansible/openshift' >> /etc/ansible/ansible.cfg
+
+COPY oc-login.sh entrypoint.sh /usr/bin/
+
+RUN useradd -u 1001 -r -g 0 -M -b /opt/apb -s /sbin/nologin -c "apb user" apb
```
Comment (fabianvf) [inline apb-base/Dockerfile:35]: Also just want to make sure that this won't break when the same useradd command is run in one of the Dockerfiles based on this one. If that will break it, would you mind updating the other Dockerfiles (postgres and postgres-demo were just added) for the sample apps to remove this? If not then :shipit: and I will update them later.

Comment (rthallisey) [reply apb-base/Dockerfile:35]: Sure I'll update it.

**Comment by fabianvf** [comment]
ACK

**Comment by rthallisey** [comment]
merging since it has an ack

**Pull request merged**